app/javascript/packs/leafletMap.js

```diff
@@ -278,14 +250,36 @@ function setupGeoJsonFeature(feature, layer) {
   label.closeTooltip();
 }
 
-async function loadGeoJsonFile(filename) {
+function setupGeoJsonFeatureOutdoor(feature) {
+  if(feature.properties.type === "uni-potsdam-building") return;
+  const marker = {
+    latlng: feature.properties.letter_coordinate.reverse(),
+    divIcon: {
+      html: feature.properties.letter,
+      className: "building-icon"
+    }
+  };
+  addMarker(marker);
+}
+
+function getBuildingStyle(feature) {
+  return {className: "building " + feature.properties.type};
+}
+
+function getRoomStyle() {
+  return {className: "hpi-room"};
+}
+
+async function loadGeoJsonFile(filename, featureCallback, styleCallback, addToMap = false) {
```

Comment:
note: the `addToMap` is a bit out of line, most other functions here take a map as an argument and only default to the global `map` object whereas this one only allows the global one; this might be fine since we only use it in the setup function though

Reply:
Also, as the L.geoJSON function creates an additional layer, you would only ever want to add this directly to the map, not to any other layer. I also tried moving the `addTo` into the `setupMap` function, but because the function call is wrapped into a `Promise.all`, that doesn't work either.